tools/metrics/tlscheck/tls.go

```diff
@@ -0,0 +1,45 @@
+package tlscheck
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"strings"
+)
+
+func GetCertificate(addr string) (*x509.Certificate, error) {
+	conn, err := tls.Dial("tcp", addr, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	for _, chain := range conn.ConnectionState().VerifiedChains {
+		for _, cert := range chain {
+			if !cert.IsCA {
```

Comment:
Doing this makes two assumptions:

* the first `VerifiedChains` element is the one we want (it probably is)
* the certificate we want to test is the first in the slice of certificates. I can't find evidence of this in the docs.

https://golang.org/pkg/crypto/tls/#ConnectionState

Reply:
The above points are probably not a problem, but I would feel better if I had the answers.